lapack/tpls/KokkosLapack_gesv_tpl_spec_avail.hpp

```diff
@@ -88,24 +76,32 @@ KOKKOSLAPACK_GESV_TPL_SPEC_AVAIL_MAGMA(Kokkos::complex<double>,
 KOKKOSLAPACK_GESV_TPL_SPEC_AVAIL_MAGMA(Kokkos::complex<float>,
                                        Kokkos::LayoutLeft, Kokkos::CudaSpace)
 
-/*
-#if defined (KOKKOSKERNELS_INST_DOUBLE) \
- && defined (KOKKOSKERNELS_INST_LAYOUTRIGHT)
- KOKKOSLAPACK_GESV_TPL_SPEC_AVAIL_MAGMA( double, Kokkos::LayoutRight,
-Kokkos::CudaSpace) #endif
-#if defined (KOKKOSKERNELS_INST_FLOAT) \
- && defined (KOKKOSKERNELS_INST_LAYOUTRIGHT)
- KOKKOSLAPACK_GESV_TPL_SPEC_AVAIL_MAGMA( float, Kokkos::LayoutRight,
-Kokkos::CudaSpace) #endif
-#if defined (KOKKOSKERNELS_INST_KOKKOS_COMPLEX_DOUBLE_) \
- && defined (KOKKOSKERNELS_INST_LAYOUTRIGHT)
- KOKKOSLAPACK_GESV_TPL_SPEC_AVAIL_MAGMA(
-Kokkos::complex<double>,Kokkos::LayoutRight, Kokkos::CudaSpace) #endif
-#if defined (KOKKOSKERNELS_INST_KOKKOS_COMPLEX_FLOAT_) \
- && defined (KOKKOSKERNELS_INST_LAYOUTRIGHT)
- KOKKOSLAPACK_GESV_TPL_SPEC_AVAIL_MAGMA( Kokkos::complex<float>,
-Kokkos::LayoutRight, Kokkos::CudaSpace) #endif
-*/
+#endif
+
+#ifdef KOKKOSKERNELS_ENABLE_TPL_ROCSOLVER
+
+#define KOKKOSLAPACK_GESV_TPL_SPEC_AVAIL_ROCSOLVER(SCALAR, LAYOUT, MEMSPACE) \
+  template <>                                                                \
+  struct gesv_tpl_spec_avail<                                                \
+      Kokkos::HIP,                                                           \
+      Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<Kokkos::HIP, MEMSPACE>,  \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                \
+      Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<Kokkos::HIP, MEMSPACE>,  \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                \
+      Kokkos::View<int*, LAYOUT, Kokkos::Device<Kokkos::HIP, MEMSPACE>,      \
```

Comment:
Should this use ``rocblas_int``? The IPIV raw pointer in the tpl_spec_decl is reinterpret_casted to that.